src/pages/index.js

```diff
@@ -64,8 +158,49 @@ document.querySelector(".profile__button-add").addEventListener("click", () => {
 });
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card", () => {
-    pictureModal.open(card.getCardData());
-  });
-  return card.getCardElement();
+  const card = new Card(
+    cardData,
+    userId,
+    "#card",
+    () => {
+      pictureModal.open(card.getCardData());
+    },
+    (cardId) => {
+      deleteModal.open();
+      deleteModal.setSubmitHandler(() => {
+        deleteModal.changeButtonText("Deleting...");
+        api
+          .deleteCard(cardId)
+          .then(() => {
+            card.deleteCard();
+          })
+          .catch((err) => {
+            console.error("Error. The request has failed: ", err);
+          })
+          .finally(() => {
+            deleteModal.changeButtonText("Yes");
+            deleteModal.close();
```

Comment:
` deleteModal.close();`

You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server)